website/content/docs/k8s/connect/ingress-controllers.mdx

```diff
@@ -0,0 +1,79 @@
+---
+layout: docs
+page_title: Ingress Controller Integrations
+description: Configuring Ingress Controllers With Consul On Kubernetes
+---
+
+# Configuring Ingress Controllers with Consul on Kubernetes
+
+-> This topic requires familiarity with [Ingress Controllers](https://kubernetes.io/docs/concepts/services-networking/ingress-controllers/)
+
+-> The following information requires Consul 1.10+, Consul-k8s 0.26+, Consul-helm 0.32+ configured with
+[Transparent Proxy](/docs/connect/transparent-proxy) mode enabled.
+
+This page describes a general approach for integrating Ingress Controllers with Consul on Kubernetes to secure traffic from the Controller
+to the backend services. This allows Consul to transparently secure traffic from the ingress point through the entire traffic flow of the service.
+
+A few steps are generally required to enable an Ingress controller to join the mesh and pass traffic through to a service:
+
+* Enable connect-injection via an annotation on the Ingress Controller's deployment: `consul.hashicorp.com/connect-inject` is `true`.
+
+* Using the following annotations on the Ingress controller's deployment, setup exclusion rules for it's ports.
+  * [`consul.hashicorp.com/transparent-proxy-exclude-inbound-ports`](https://www.consul.io/docs/k8s/connect#consul-hashicorp-com-transparent-proxy-exclude-inbound-ports) - Provides the ability to exclude a list of ports for
+inbound traffic that the service exposes from redirection. Typical configurations would require all inbound service ports
+for the controller to be included in this list.
+  * [`consul.hashicorp.com/transparent-proxy-exclude-outbound-ports`](https://www.consul.io/docs/k8s/connect#consul-hashicorp-com-transparent-proxy-exclude-outbound-ports) - Provides the ability to exclude a list of ports for
```

Comment:
So if I was hitting the kube api on `10.0.0.1:443` could I use either `consul.hashicorp.com/transparent-proxy-exclude-outbound-ports: 443` or `consul.hashicorp.com/transparent-proxy-exclude-outbound-cidrs: 10.0.0.1`? Just want to know how exactly this works.

Reply:
Either would work to enable the Kube API to be directly accessed from the pod. Using `exclude-outbound-ports` would permit access to _any_ destination IP over port 443, whereas `exclude-outbound-cidrs` would permit access to 10.0.0.1 on any destination port.

Here's the rendered iptables rules for reference.

```shell
$ iptables --table nat --list-rules OUTPUT
-P OUTPUT ACCEPT
-A OUTPUT -p tcp -j CONSUL_PROXY_OUTPUT

$ iptables --table nat --list-rules CONSUL_PROXY_OUTPUT
-N CONSUL_PROXY_OUTPUT
-A CONSUL_PROXY_OUTPUT -m owner --uid-owner 996 -j RETURN
-A CONSUL_PROXY_OUTPUT -d 10.0.0.1/32 -j RETURN
-A CONSUL_PROXY_OUTPUT -p tcp -m tcp --dport 443 -j RETURN
-A CONSUL_PROXY_OUTPUT -d 127.0.0.1/32 -j RETURN
-A CONSUL_PROXY_OUTPUT -j CONSUL_PROXY_REDIRECT
```